finality-provider/service/fp_instance.go

```diff
@@ -116,12 +116,22 @@ func (fp *FinalityProviderInstance) Start() error {
 		zap.String("pk", fp.GetBtcPkHex()), zap.Uint64("height", startHeight))
 
 	poller := NewChainPoller(fp.logger, fp.cfg.PollerConfig, fp.cc, fp.consumerCon, fp.metrics)
+	fp.poller = poller
 
-	if err := poller.Start(startHeight); err != nil {
-		return fmt.Errorf("failed to start the poller: %w", err)
+	// get the last finalized height
+	lastFinalizedBlock, err := fp.latestFinalizedBlockWithRetry()
+	if err != nil {
+		return err
 	}
 
-	fp.poller = poller
+	// Start the poller if fast sync is disabled or there's no finalized block
+	if (fp.cfg.FastSyncInterval == 0 || lastFinalizedBlock == nil) && !fp.poller.IsRunning() {
```

Comment:
without `lastFinalizedBlock == nil`, babylon e2e `TestFastSync` will fail b/c `tm.WaitForFpVoteCast(t, fpIns)` will timeout

this is b/c in the code below, `tryFastSync` will not do anything

<img width="977" alt="image" src="https://github.com/user-attachments/assets/e90bcb2d-f605-45f4-ac6b-5459c9100cfb">

Reply:
Can we move `!fp.poller.IsRunning()` out to a new if condition? imo if `!fp.poller.IsRunning()` then we need to `return err`? Also if this if statement is false, when will the poller start?

Reply:
If this `if` statement is `false`, we start the poller after fast sync is finished.
 ```Go
 // start poller after fast sync is finished
if !fp.poller.IsRunning() {
	err := fp.poller.Start(res.LastProcessedHeight + 1)
	if err != nil {
		fp.logger.Error("failed to start the poller", zap.Error(err))
		fp.reportCriticalErr(err)
	}
	continue
}
 ```

Reply:
Not sure we should add these check because up to here the fast sync is already finished and should be safe to start the poller. Can we have a workaround in e2e?